### PR TITLE
PTSD: Now standard issue for all heads of staff

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request

image
image

All heads may acquire their PTSD by checking their locker for it, except for the HoS who starts the round with PTSD regardless of choice. The Captain and HoS may also find that their full size Energy Guns have been swapped out for PTSD (Not their unique guns, they still have these) and the other heads will find that their telescopic batons are no longer capable of inflicting stamcrit. Telescopic batons are still useful for tripping and disarming. 

The gun in question is a modified version of the HoP's existing miniature energy gun. Modifications include 3 fewer disabler shots per charge and slow self-charging capabilities at the expense of being able to use recharging stations. Additionally this adds and evenly spreads some additional equipment which will be desirable theft targets for antagonists, especially if any specific heads of staff are missing and their locker is unguarded. 

<details>
<summary>Full description of the gun</summary>
Has a flashlight built in, 12 disabler shots per full charge, 6 lethal shots per full charge, 120 seconds to reach full charge (regaining 2 disabler or 1 lethal every ten seconds). The gun may not use recharges and must slowly recharge on its own. 
</details>

## Why It's Good For The Game:
Even with most shifts being canonically peaceful, Nanotrasen is well aware that it does have enemies and it makes sense that all upper level members of staff would be issued a small arm for self-defense. This helps bring the heads of staff that otherwise only really had a baton before (CE, CMO, RD) up to a more reasonable level of self-defense. 

The telescopic baton has has lost its ability to inflict stamcrit to make up for the increased overall firepower. However this also means that heads can no longer keep someone in stamcrit indefinitely since the gun quickly runs out of charge which better reinforces that both tools are for defensive use, not offensive. Baton is still useful for tripping and disarming for those robust enough to target limbs with it, and its description has been updated to make its capabilities more intuitive to those who inspect it. 

It also helps bring the Captain and HoS starting power level down to a more reasonable level without forbidding them from access to their literal starting equipment and it gives antagonists a new avenue to acquire some basic and concealable weaponry, and while it has limited uses, they won't have to find a sneaky way to charge or reload.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Weapon performs as described: 12 disabler, 6 lethal, no visual bugs when ammo is empty regardless of what combination of shots is used, recharges once every 20 seconds. Flashlight works. 

Weapon appropriately spawns in all five lockers it is supposed to and in the suit slot of the HoS. 

</details>

## Changelog
:cl: Buggy123, St0rmC4st3r, Rukofamicom
add: Adds PTSD - a slightly modified version of the miniature energy gun with 3 fewer disabler shots per charge, but the ability to fully recharge itself in 120 seconds. 
balance: All heads of staff can now pick up PTSD at the start of the round by checking their lockers, except for the HoS who starts with PTSD whether they want it or not.
balance: The HoS and Captain no longer have roundstart access to a full size energy gun due to acquiring PTSD. They do still have their unique weapons. 
balance: To make up for the increased firepower, the telescopic baton is no longer deals stamina damage, except when targeting arms. It is still effective for tripping and disarming, but not locking someone down in place indefinitely.
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
I expect the recharge rate might need to be adjusted; 60 seconds is pretty fast for full charge, but I don't think we'll know for sure without testing and the gun itself has a pretty low burst potential as it stands.

**Comment by St0rmC4st3r** [comment]
Uuuuh, more PTSD? #1188

**Comment by Rukofamicom** [comment]
Yes, that's where I stole the name from. However this gun is *actually* implemented while the qwerty approved gun sits completely unused. 

you snooze you lose

**Comment by St0rmC4st3r** [comment]
I added it because I liked the sprite and didn't want it to go to waste

**Comment by Rukofamicom** [comment]
it is a very nice sprite

You know what, I'm gonna use it.

**Comment by Rukofamicom** [comment]
Most recent merge did not, in fact, fix flashlight code. No separate PR is coming for that yet. Need to sleep before I keep trying to squish that bug.

**Comment by Mat05usz** [comment]
I like it, CE/CMO/RD really lacked any reasonable self-defense tool given how valuable targets they are.

**Comment by Coding-Grape** [comment]
> I like it, CE/CMO/RD really lacked any reasonable self-defense tool given how valuable targets they are.

Advanced batong on legs, apply toolbox/saw/welder/other robust object

**Comment by Rukofamicom** [comment]
> Advanced batong on legs

You reminded me that I meant to take this away in favor of the PTSD for most heads, but I got caught up in meming with the name and forgot. 

The telescopic baton is a very good item that... unfortunately isn't intuitive to learn how to use appropriately. Until someone is more or less explicitly told how to use it, it appears to be a nearly useless stick. On the other hand, everyone knows how to use an energy gun almost immediately even the first time they pick it up. The only thing that doesn't come immediately is switching firing modes and flashlight toggle.

**Comment by DatBoiTim** [comment]
I don't give a care what this PR does,

MERGE IT

**Comment by Mat05usz** [comment]
> The telescopic baton is a very good item that... unfortunately isn't intuitive to learn how to use appropriately. Until someone is more or less explicitly told how to use it, it appears to be a nearly useless stick. On the other hand, everyone knows how to use an energy gun almost immediately even the first time they pick it up. The only thing that doesn't come immediately is switching firing modes and flashlight toggle.

So you take a melee self-defense weapon, which was original and fun to use, and replace it with a ranged 'pew pew' cause Bee players have become less robust? Now I dislike this PR.

>  All heads of staff can now pick up PTSD at the start of the round by checking their lockers

You mean all players unaware of changes can now get banned for powergaming more easily?

Edit:
I can already see those 'as CE, for no reason had PTSD' banned by Host. Expire date: Never

**Comment by Rukofamicom** [comment]
> as CE, for no reason had PTSD

The intent of this PR is to supply them with those weapons from the start. I've said this PR will need admin approval to see the light of day.

> So you take a melee self-defense weapon, which was original and fun to use, and replace it with a ranged 'pew pew' cause Bee players have become less robust?

Unfortunately, yes. I dislike unintuitive game design. Players shouldn't need to be told that they should first target the legs and then the chest to properly use their stick with no apparent effect on the target or any meaningful description that leads you to realize it's actually a better and infinite stun baton in practice. 

That and even with how useful the telescopic baton is, it just looks lame even if you know about its secrets. Pew pew laser gun much cooler and more fitting.

**Comment by EvilDragonfiend** [comment]
>  (regaining 2 disabler or 1 lethal every ten seconds)
twenty seconds, right?

**Comment by Rukofamicom** [comment]
> twenty seconds, right?

Only as of very recent commits, I was in the process of updating the PR description still.

**Comment by Rukofamicom** [comment]
#### After feedback from discord and otherwise:

* General reception is that the guns were at least a bit too strong and capable for their intended purpose.
* The telescopic baton will be very sorely missed by those who did know how to properly utilize it
* I can easily make the telescopic baton more intuitive by updating its description

#### So I have changed:

* PTSD now has double the recharge rate it did. 20 seconds per recharge cycle which grants 2 more disabler and 1 more lethal. 
* PTSD can no longer quick charge via charging stations - it is only possible to slow charge passively. Once its rounds are spent and you are safe, you should be seeking assistance not trying to ready up for further combat. 
* Re-added telescopic baton to all of the heads, but removed its stamcrit capabilities to make up for the added firepower in the form of a gun, but also to reinforce its intended use as a defensive weapon rather than an offensive one. It may no longer be used to permanently lock someone in place via stamcrit. 
* Telescopic baton description now indicates that the user is intended to target hands and feet to get the most out of it.

#### Further potential changes
* Removing lethal mode from the PTSD entirely
* Reduce PTSD capacity by 33% - 4 lethals isn't enough to crit by itself then, and 8 disabler is still probably enough for taking down one target unless they're roided up. 

Ultimately I think this is balanced well how it is now and will be waiting for maintainer thoughts and instructions before making any further modifications to the PR. Assuming the idea as a whole is moderately well receieved, I believe a testmerge is the only meaningful way to improve upon the balance of the PR at this point instead of continued hypotheticals.

**Comment by Rukofamicom** [comment]
Slapped the minor refactor (for how guns with unusual charge levels are done) from #6753 into this one so there won't be an eventual conflict if/when one gets merged

**Comment by Rukofamicom** [comment]
oh fuck what the hell did I break by trying to add comments lmao

**Comment by Rukofamicom** [comment]
Updated description to include credit where it's due: @St0rmC4st3r for the pun and generally trying to bring the gun into the game first, and @Buggy123 for the really nice base sprite. 

The PR may have started as a mini_egun, but it's something of a conglomeration now.

**Comment by Isy232** [comment]
> balance: The HoS and Captain no longer have roundstart access to a full size energy gun due to acquiring PTSD. They do still have their unique weapons.

Maybe leave the HoS and Captains normal energy guns? The captain and HoS are often in not just self defense situations but dealing with station threats so replacing their normal gun with one almost entirely meant for self defense seems a little odd.

I like the new gun, but I also have a few concerns over balance changes. If every head can act as a mini-security for their department with a mini-disabler, it might be significantly harder for cult and revs. This is the sort of thing that should probably be tested in a test merge.

Do you also need to nerf the telescopic baton? Its alright, especially if you can get the limb hits off, but you never see it used massively due to the stun baton just being better in almost all cases. Perhaps it could be left alone?

**Comment by Rukofamicom** [comment]
> Maybe leave the HoS and Captains normal energy guns? The captain and HoS are often in not just self defense situations but dealing with station threats so replacing their normal gun with one almost entirely meant for self defense seems a little odd.

Currently, neither should be carrying full-size energy guns on Green alert period. This is a much more lore-friendly gun for them to tote around when there isn't a threat than the energy equivalent of an assault rifle which admins who enforce SOP currently expect them not to be keeping on their person. 
*HoS literally spawns with a current level of equipment that SOP forbids*. 
Captain at least has to go to his locker and display case first (display case not anymore now), but it's bad for them to be right within reach when players are told not to take them. That's why I made the captain's laser PR as well.

> I like the new gun, but I also have a few concerns over balance changes. If every head can act as a mini-security for their department with a mini-disabler, it might be significantly harder for cult and revs. This is the sort of thing that should probably be tested in a test merge.

**I would have a cow if this was straight merged without testing, even with it being my own PR**. It is absolutely in need of testing, there's only so much theorizing anyone can do on paper alone. 

That said, I'd like to remind you that all heads, for most of this server's history, heads had a full-size 25 shot disabler as part of their standard gear. This was taken and replaced with a better version of security's stun baton.

> Do you also need to nerf the telescopic baton? Its alright, especially if you can get the limb hits off, but you never see it used massively due to the stun baton just being better in almost all cases. Perhaps it could be left alone?

See above. It is literally an upgraded, infinite stun baton. This wild misconception and not knowing how to use it without being told are the primary reasons it doesn't see use, not because it is weak.  Keep in mind that stamcrit occurs at 100 stamina damage and compare them real quick:

Stun baton:
```
Does not fit in pocket
10 uses before needing to recharge
75 stamina per hit
Will disarm on upper limb use, disabling the limb with stamina damage in one hit
Will slow on leg use, but not knock down
```

Telescopic baton:
```
Pocket sized
Infinite uses
55 Stamina Per hit
Will disarm on upper limb use, disabling the limb with stamina damage in one hit
Will not do stamina damage to legs, but knockdown in one hit
```

**Comment by Isy232** [comment]
> Currently, neither should be carrying full-size energy guns on Green alert period. This is a much more lore-friendly gun for them to tote around when there isn't a threat than the energy equivalent of an assault rifle...

Yeah, fair enough. I agree, I was more concerned about the possible balance changes by nerfing the cap and HoS. However, as you agree, this is the kind of thing that needs to be figured out by testing.

> See above. It is literally an upgraded, infinite stun baton...

I suppose my thoughts are that the telescopic baton is worse comes from its lower damage. Against armoured opponents, the telescopic baton would require quite a few more hits to take down your opponent in comparison to the stun baton.

Also, the telescopic baton would not inflict stamina slowdown upon its first hit, which means your opponent can run away. And if they run away, they only need to regenerate 10 stamina before it would take 3 total hits to take them down.

I don't agree that its a direct upgrade, the lack of stamina slowdown on the first hit make it more useful as a self defense weapon by tripping or disarming your attacker so you may run away. If you are using this weapon offensively, it is noticably weaker then the stun baton, especially against someone with a little bit of armour.

I still think it should just be left alone, but it would also be rather akward to have a "baton" used for disarming and tripping only, perhaps just removing it would be better.

**Comment by Rukofamicom** [comment]
> Also, the telescopic baton would not inflict stamina slowdown upon its first hit

This happens at 40 stamina damage doesn't it? 

> I still think it should just be left alone, but it would also be rather akward to have a "baton" used for disarming and tripping only, perhaps just removing it would be better.

This is what I initially did. I re-added the non-stamina effects at the request of @Mat05usz because even without stamina these are pretty useful effects.

**Comment by Isy232** [comment]
> This happens at 40 stamina damage doesn't it?

I thought it was 60. I could be wrong but I dont have time to test now.

> This is what I initially did. I re-added the non-stamina effects at the request of @Mat05usz because even without stamina these are pretty useful effects.

Fair enough, this could be figured out in a TM.

File: code/modules/projectiles/guns/energy/energy_gun.dm
Comment (PowerfulBacon) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: I think it should be named something more suitable, like the full name 'Personal Tiny Self Defense' or just Personal Energy Gun or something

Comment (Rukofamicom) [reply code/modules/projectiles/guns/energy/energy_gun.dm]: image

**Review by ike709 — APPROVED** [review]

**Comment by Oscari-pog** [comment]
After a few rounds of playing with this, I can no longer say that I support this PR as is only incentivizes validhunting and makes it easier.

**Comment by Rukofamicom** [comment]
Can you elaborate what you mean and give some examples that led you to this?

**Comment by Rukofamicom** [comment]
Can you elaborate what you mean and give some examples that led you to this?
@Oscari-pog

**Comment by Oscari-pog** [comment]
> Can you elaborate what you mean and give some examples that led you to this?

Observing how heads now react to threats, they no longer seek to escape the situation, or approach it cautiously. They now immediately pull their weapon, and in large amount of cases, they default to lethals. This is not being utilized as a self-defense weapon by the current command players.

I personally watched a head pull lethals on a greytider in their department, wordlessly. Security was called only after they were in crit due to the easily concealable lasergun (that, by the way, doesnt shoot weakened lasers as my impression initially was.).

If the direction we want to go through, which would ideally be the head calling security and not rushing any antag with a weapon, giving them another weapon in addition to their baton is not the way.

If you give people a weapon, they will immediately use it for any reason that you give them. That is ultimately the bane of arming every head, as they will now act like a officer, only worsening the fact that many of them already acted like one even without the gun.

**Comment by Rukofamicom** [comment]
First taken from SOP, second from the actual rules:
> 1) Command Staff are encouraged to request the assistance of Security when purging undesirables from their departments.
> 2) Heads of staff are permitted to defend their own department and subordinates, but should not be patrolling the station for threats unless there is no security.

Would you say they are violating this? Part of the reasoning behind PTSD was enabling them to actually be capable of defending their department, which the baton barely enabled. Heads of staff **are** combat trained just like security is and are responsible for defending their work space unlike their subordinates. 

Heads of staff are meant to be a threat, like security is minus patrolling, but they were not equipped to actually be one before. 

> that, by the way, doesnt shoot weakened lasers as my impression initially was.

Not weakened, just very few of them if it's set to lethal. You can stamcrit four targets, but only fullcrit one and only if you're very accurate. 

@Oscari-pog (Pinging to make sure you see response)

**Pull request merged**
